tests/test_task.py

```diff
@@ -0,0 +1,132 @@
+"""
+Task Test Suite
+"""
+
+import unittest
+
+from pydantic import Field, ValidationInfo, field_validator
+
+import aind_behavior_curriculum as abc
+
+
+class ExampleTask(abc.Task):
+    """
+    Example Task
+    """
+
+    # Required: Define type annotations for strict type checks.
+    # Optional: Make fields immutable with the 'frozen=True'.
+    field_1: int = Field(0)
+    field_2: int = Field(0)
+    field_3: float = Field(0.5)
+    field_4: float = Field(0.5)
+    field_5: str = Field("Immutable Field", frozen=True)
```

Comment:
While i am not necessarily against using frozen, I think we should keep in mind that the spec is defined in the json-schema and not in pydantic. As a result, make sure that you push as much validation to the json side as possible. AFAIK, `frozen` does not get rendered to the schema. You should probably just declare the property as:

```python
field_5: Literal["Immutable Field"] = Immutable Field
```